go-controller/pkg/metrics/cluster_manager.go

```diff
@@ -117,6 +117,11 @@ func RegisterClusterManagerFunctional() {
 		prometheus.MustRegister(metricEgressIPRebalanceCount)
 		prometheus.MustRegister(metricEgressIPCount)
 	}
+	if err := prometheus.Register(MetricResourceRetryFailuresCount); err != nil {
```

Comment:
@npinaeva yes, this is needed for microshift case where both control plane and node run with same pod. 
Do you see a harm with `Unregister` ?

Reply:
I don't really know how Unregister works, but if there are multiple processes using this metric, then there is a chance to Unregister it from the cluster-manager while some other component is still using it. Maybe @martinkennelly knows more about how it works?
Also now I see we only Unregister metrics for cluster-manager, but not for any other components, and I am not sure why.

Reply:
sounds good, let's maybe have another pr to remove it?

Reply:
yes @npinaeva , i talked to @martinkennelly and would propose another PR to remove unregister code.

Reply:
raised a PR https://github.com/ovn-org/ovn-kubernetes/pull/3826 for removing unregister, will update this PR once that is merged.

Reply:
No @jcaamano , this metric is also registered in `ovnkube-controller` and `ovnkube-node`  modules , all would be run on same pod in case of microshift.  hence we needed this change.